src/cargo/ops/cargo_new.rs

```diff
@@ -816,18 +816,24 @@ fn mk(config: &Config, opts: &MkOptions<'_>) -> CargoResult<()> {
         // This should not block the creation of the new project. It is only a best effort to
         // inherit the workspace package keys.
         if let Ok(mut workspace_document) = root_manifest.parse::<toml_edit::Document>() {
+            let (display_path, exclude) = get_display_path_and_check_exclusion(
+                &root_manifest_path,
+                &path,
+                &workspace_document,
+            )?;
             if let Some(workspace_package_keys) = workspace_document
                 .get("workspace")
                 .and_then(|workspace| workspace.get("package"))
                 .and_then(|package| package.as_table())
             {
-                update_manifest_with_inherited_workspace_package_keys(
-                    opts,
-                    &mut manifest,
-                    workspace_package_keys,
-                )
+                if !exclude {
```

Comment:
Why is the `!exclude` check inside of the `if`?  Seems like from the perspective of trying to describe this code, we shouldn't try to look stuff up in the workspace if this is a workspace member.

Putting it on the outside will also make it easier to avoid bugs as we add new sections to inherit

Reply:
Moved out.